### PR TITLE
Implement \JsonSerializable for Scopes

### PR DESCRIPTION
**Pull request by remotelyliving**
So the idea here may have already been a thing...but I'd like to propose implementing \JsonSerializable on Scopes.

It saves a call to `toArray()` and would allow for Scopes to be passed directly to `json_encode()`

This saves a little time depending on how mocky / stubby your tests get in your consumer implementation as there is one less message to pass through with whatever value you're interested in returning / asserting against. 

So in our codebase (and I assume some others) our controller or responder might look like

```php

$this->respond($this->manager->createData($resource)->toArray());

// OR

$this->respond($this->manager->createData($resource));

// to test with mocks we have to mock a Scope and stub `toArray()` and assert on that
// or we could just assert that the responder method is is called with the mock
```

Basically it's one less part of the message chain here and can hopefully make people's lives a little easier. Could totally be wrong though! Not everyone uses this the same way. 

Either way, thank you contributors for making such a great library that solves some highly obvious pain-points that are rarely addressed in php api land.

### CODE REVIEW

**Comment by KorvinSzanto** [comment]
@remotelyliving This looks good and ready to merge, can you add a commit to this PR that adds a line to the `CHANGELOG.md` describing this change in a few words?

**Comment by remotelyliving** [comment]
Thanks @KorvinSzanto ! I updated the `CHANGELOG.md` accordingly. Let me know if there's anything else I can do.

**Comment by KorvinSzanto** [comment]
Thanks!
